sympy/vector/dyadic.py

```diff
@@ -225,6 +225,12 @@ def _sympystr(self, printer):
         return "({}|{})".format(
             printer._print(self.args[0]), printer._print(self.args[1]))
 
+
+    @property
+    def func(self):
+        return self.__class__
```

Comment:
I can't remember any more. Why are these `func` methods needed?

Reply:
In the other PR, you had suggested to put comments here explaining what is going on, I will put those here too. But basically, these classes do not follow the design pattern for `Basic`s to such a degree that we could not find a way to make them work without essentially rewriting them. (e.g. they are storing all sorts of weird non-`Basic` arguments)